src/agent/debuglet.ts

```diff
@@ -107,6 +113,49 @@ const formatBreakpoints = function(
           .join('\n');
 };
 
+/**
+ * CachedPromise stores a promise. This promise is initially null and can be
+ * assigned to a new promise when refresh is called. This promise will be
+ * resolved by calling member function resolve.
+ */
+export class CachedPromise {
+  private promise: Promise<void>|null = null;
+  private promiseResolve: (() => void)|null = null;
+
+  refresh(): Promise<void> {
+    this.promise = new Promise<void>((resolve) => {
+      this.promiseResolve = resolve;
+    });
+    return this.promise;
+  }
+  get(): Promise<void>|null {
+    return this.promise;
+  }
+  resolve(): void {
+    // Each promise can be resolved only once.
+    if (this.promiseResolve) {
+      this.promiseResolve();
+      this.promiseResolve = null;
+    }
+  }
+  clear(): void {
+    this.promise = null;
+  }
+}
+
+/**
+ * IsReadyManager is a wrapper class to call debuglet.isReady().
+ */
+export class IsReadyManager {
```

Comment:
I would recommend using an `IsReady` interface here and have an `IsReadyImpl` (or another name if you prefer) that implements the interface.  Then the places that are currently referencing this class, can instead reference the interface since the class details are not actually needed.  This allows the implementation to be hidden.

That is, use:
```
export interface IsReady {
  isReady(): Promise<void>;
}
```
and 
```
class IsReadyImpl implements IsReady {
  private debuglet: Debuglet;
  constructor(debuglet: Debuglet) {
    this.debuglet = debuglet;
  }

  isReady(): Promise<void> {
    return this.debuglet.isReady();
  }
}
```

Reply:
I agree. Thanks for bringing it out. I have it updated.